pkg/kv/kvserver/kvstorage: BenchmarkWAGTruncation failed

pkg/kv/kvserver/kvstorage.BenchmarkWAGTruncation [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachprodMicrobenchmarkWeeklyWip/21335848?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachprodMicrobenchmarkWeeklyWip/21335848?buildTab=artifacts#./artifacts/microbench/experiment) on master @ [9709f69a4fd6b6873937d9b3e9e27c825bb215d7](https://github.com/cockroachdb/cockroach/commits/9709f69a4fd6b6873937d9b3e9e27c825bb215d7):


```
--- FAIL: BenchmarkWAGTruncation/batchSize=4
BenchmarkWAGTruncation/batchSize=8
    wag_truncator_test.go:749: 
        	Error Trace:	pkg/kv/kvserver/kvstorage/wag_truncator_test.go:749
        	            				GOROOT/src/testing/benchmark.go:219
        	            				GOROOT/src/testing/benchmark.go:245
        	            				src/runtime/asm_amd64.s:1771
        	Error:      	Not equal: 
        	            	expected: 0x1
        	            	actual  : 0x0
        	Test:       	BenchmarkWAGTruncation/batchSize=8
--- FAIL: BenchmarkWAGTruncation/batchSize=8
BenchmarkWAGTruncation/batchSize=16
    wag_truncator_test.go:749: 
        	Error Trace:	pkg/kv/kvserver/kvstorage/wag_truncator_test.go:749
        	            				GOROOT/src/testing/benchmark.go:219
        	            				GOROOT/src/testing/benchmark.go:245
        	            				src/runtime/asm_amd64.s:1771
        	Error:      	Not equal: 
        	            	expected: 0x1
        	            	actual  : 0x0
        	Test:       	BenchmarkWAGTruncation/batchSize=16
--- FAIL: BenchmarkWAGTruncation/batchSize=16
BenchmarkWAGTruncation/batchSize=32
    wag_truncator_test.go:749: 
        	Error Trace:	pkg/kv/kvserver/kvstorage/wag_truncator_test.go:749
        	            				GOROOT/src/testing/benchmark.go:219
        	            				GOROOT/src/testing/benchmark.go:245
        	            				src/runtime/asm_amd64.s:1771
        	Error:      	Not equal: 
        	            	expected: 0x1
        	            	actual  : 0x0
        	Test:       	BenchmarkWAGTruncation/batchSize=32
--- FAIL: BenchmarkWAGTruncation/batchSize=32
BenchmarkWAGTruncation/batchSize=64
    wag_truncator_test.go:749: 
        	Error Trace:	pkg/kv/kvserver/kvstorage/wag_truncator_test.go:749
        	            				GOROOT/src/testing/benchmark.go:219
        	            				GOROOT/src/testing/benchmark.go:245
        	            				src/runtime/asm_amd64.s:1771
        	Error:      	Not equal: 
        	            	expected: 0x1
        	            	actual  : 0x0
        	Test:       	BenchmarkWAGTruncation/batchSize=64
--- FAIL: BenchmarkWAGTruncation/batchSize=64
    wag_truncator_test.go:752: -- test log scope end --
--- FAIL: BenchmarkWAGTruncation
FAIL

test logs left over in: /tmp/logBenchmarkWAGTruncation3394231623
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*BenchmarkWAGTruncation.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-63595

--- a/pkg/kv/kvserver/kvstorage/wag_truncator_test.go
+++ b/pkg/kv/kvserver/kvstorage/wag_truncator_test.go
@@ -719,6 +719,8 @@ func BenchmarkWAGTruncation(b *testing.B) {
 			b.StopTimer()
 			st := cluster.MakeTestingClusterSettings()
 			wagTruncatorBatchSize.Override(ctx, &st.SV, batchSize)
+			// Disable WAG retention to allow full WAG truncation.
+			wagSuffixRetentionCount.Override(ctx, &st.SV, 0)
 			eng := storage.NewDefaultInMemForTesting()
 			defer eng.Close()
 			engines := MakeEngines(eng)

